Has `deliveryCallback` a wrong signature?

Hello!
We are using this awesome lib for a while now.

# Context

I tried today to upgrade from `3.1.0` to `4.0.1` but I faced the following error on my project:
```
/src/Commons/Kafka/Producer.hs:56:23: error:
    • Couldn't match expected type ‘Callback’
                  with actual type ‘hw-kafka-client-4.0.1:Kafka.Internal.Setup.KafkaConf
                                    -> IO ()’
    • Probable cause: ‘deliveryCallback’ is applied to too few arguments
      In the first argument of ‘setCallback’, namely
        ‘(deliveryCallback (logError logState))’
      In the first argument of ‘(<>)’, namely
        ‘setCallback (deliveryCallback (logError logState))’
      In the second argument of ‘(<>)’, namely
        ‘setCallback (deliveryCallback (logError logState))
           <> setCallback (logCallback (handleLibrdkafkaLogs logState))’
   |
56 |       <> setCallback (deliveryCallback (logError logState))
   |
```

# Research 
I did some research on the documentation, but did not find anything interesting, so I took a look on the code.

Without success and the same error, I also tried:
```haskell
setCallback (deliveryCallback print)
```

I found out [this PR](https://github.com/haskell-works/hw-kafka-client/pull/161) add the 
```haskell
newtype Callback = Callback (KafkaConf -> IO ())
```

But I also found out the `deliveryCallback` (that is leading to the previous error) still have the following signature:
```haskell
deliveryCallback :: (DeliveryReport -> IO ()) -> KafkaConf -> IO ()
```

# Conclusion

First, I really hope I did not miss anything in the documentation to upgrade from `3.1.2` to `4.0.1`!

Is the `Callback` wrapper the real responsible of the error I am currently facing?

## Bonus
I thought I would be able to fix this by myself, so in `Kafka.Producer.Callbacks` I tried:
```haskell
deliveryCallback :: (DeliveryReport -> IO ()) -> Callback
deliveryCallback callback = Callback $ \kc -> rdKafkaConfSetDrMsgCb (getRdKafkaConf kc) realCb
```

But then I get stuck on `Kafka.Producer` (line122) when:
```haskell
deliveryCallback (const mempty) kc
```
I am wondering if this is really necessary... Because in `Kafka.Producer.ProducerProperties`
```haskell
setCallback :: Callback -> ProducerProperties
setCallback cb = mempty { ppCallbacks = [cb] }
```


I would be happy and proud if I was able to help this project!

Thanks again!

--- a/src/Kafka/Producer.hs
+++ b/src/Kafka/Producer.hs
@@ -117,7 +117,7 @@ newProducer pps = liftIO $ do
   tc <- topicConf (TopicProps $ (ppTopicProps pps))
 
   -- add default delivery report callback
-  deliveryCallback (const mempty) kc
+  let Callback setCb = deliveryCallback (const mempty) in setCb kc
 
   -- set callbacks
   forM_ (ppCallbacks pps) (\(Callback setCb) -> setCb kc)

--- a/src/Kafka/Producer.hs
+++ b/src/Kafka/Producer.hs
@@ -117,7 +117,8 @@ newProducer pps = liftIO $ do
   tc <- topicConf (TopicProps $ (ppTopicProps pps))
 
   -- add default delivery report callback
-  let Callback setCb = deliveryCallback (const mempty) in setCb kc
+  let Callback setDeliveryCallback = deliveryCallback (const mempty)
+  setDeliveryCallback kc
 
   -- set callbacks
   forM_ (ppCallbacks pps) (\(Callback setCb) -> setCb kc)

--- a/src/Kafka/Producer/Callbacks.hs
+++ b/src/Kafka/Producer/Callbacks.hs
@@ -15,7 +15,7 @@ import           Foreign.StablePtr      (castPtrToStablePtr, deRefStablePtr, fre
 import           Kafka.Callbacks        as X
 import           Kafka.Consumer.Types   (Offset(..))
 import           Kafka.Internal.RdKafka (RdKafkaMessageT(..), RdKafkaRespErrT(..), rdKafkaConfSetDrMsgCb)
-import           Kafka.Internal.Setup   (KafkaConf(..), getRdKafkaConf)
+import           Kafka.Internal.Setup   (KafkaConf(..), getRdKafkaConf, Callback(..))
 import           Kafka.Internal.Shared  (kafkaRespErr, readTopic, readKey, readPayload)
 import           Kafka.Producer.Types   (ProducerRecord(..), DeliveryReport(..), ProducePartition(..))
 import           Kafka.Types            (KafkaError(..), TopicName(..))
@@ -27,8 +27,8 @@ import           Kafka.Types            (KafkaError(..), TopicName(..))
 --   callbacks. For callbacks to individual messsages see
 --   'Kafka.Producer.produceMessage\''./
 --
-deliveryCallback :: (DeliveryReport -> IO ()) -> KafkaConf -> IO ()
-deliveryCallback callback kc = rdKafkaConfSetDrMsgCb (getRdKafkaConf kc) realCb
+deliveryCallback :: (DeliveryReport -> IO ()) -> Callback
+deliveryCallback callback = Callback $ \kc -> rdKafkaConfSetDrMsgCb (getRdKafkaConf kc) realCb
   where
     realCb :: t -> Ptr RdKafkaMessageT -> IO ()
     realCb _ mptr =

